scalameta/semantic/shared/src/main/scala/scala/meta/semantic/Message.scala

```diff
@@ -0,0 +1,33 @@
+package scala.meta
+package semantic
+
+import org.scalameta.adt._
+import org.scalameta.data._
+import org.scalameta.unreachable
+
+@data class Message(anchor: Anchor, severity: Severity, message: String) {
```

Comment:
The problem with `Position.Range` is that it is tied to an `Input`, and that's a bit more than just `AbsolutePath`.

upd. Oh wow, actually `Input` fits perfectly here. It doesn't have to be `Input.File` that simply wraps an absolute path, but can also be `Input.LabeledString` in which case it will embed the contents. While the usefulness of the latter is debatable (see #777), the flexibility of `Position.Range` is clear.